### PR TITLE
CI: Re-group PHPUnit jobs.

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

- Use lightweight build for REST API tests (building assets is not required; -2 mins of build time)
- Re-group PHPUnit-related jobs (-1 job)

### Testing

- All CI checks shall pass
- It should be only 4 PHP-jobs

### CODE REVIEW

**Review by adimoldovan — CHANGES_REQUESTED** [review]
> Skip Gutenberg-related E2E tests in core E2E (covered by release and daily checks; -3 jobs)

Those tests are not targeting Gutenberg specifically, and they need to run on PRs. They are tagged with `gutenberg` because they cover flows affected by GB being installed or not - that's why we run them daily in different setups. But we need them to run in the default setup.

> Re-group PHPUnit related jobs and leverage PHP 8.* for better tests performance where possible (-1 job)

On the unit tests, the PHP version is chosen by the coverage we want to have rather than performance. We had the discussion of switching from PHP 7.4 as default, I think we want do that instead.

**Comment by kalessil** [comment]
> Those tests are not targeting Gutenberg specifically, and they need to run on PRs. They are tagged with gutenberg because they cover flows affected by GB being installed or not - that's why we run them daily in different setups. But we need them to run in the default setup.

I see, I'm going to revert the changes in sharding.

Edit: reverted in 4416be96df58b77280ed187447fdca5bca1b7a0a

**Comment by kalessil** [comment]
> On the unit tests, the PHP version is chosen by the coverage we want to have rather than performance. We had the discussion of switching from PHP 7.4 as default, I think we want do that instead.

So, it's PHP 8 on the latest WordPress and L-2/L-1/Nightly on PHP 7.4?

**Comment by adimoldovan** [comment]
> > On the unit tests, the PHP version is chosen by the coverage we want to have rather than performance. We had the discussion of switching from PHP 7.4 as default, I think we want do that instead.
> 
> So, it's PHP 8 on the latest WordPress and L-2/L-1/Nightly on PHP 7.4?

Yeah, it looks like it.

The default PHP comes from [.wp-env.json](https://github.com/woocommerce/woocommerce/blob/c269a86efc004c1320c74af0c58613e2382a3836/plugins/woocommerce/.wp-env.json#L2), which is now set to 7.4. So it's default for every job that has no `testEnv.config.phpVersion` configured differently.
I agree we could combine the PHP with WP versions to reduce the number of jobs without affecting coverage.

**Comment by kalessil** [comment]
Thanks! Updated in 2c920e8861cbf6f72b7c851c269ca14de36ac773.
We are ready for the next review round.

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -155,7 +132,7 @@
 					]
 				},
 				{
-					"name": "PHP WP: latest - 1",
+					"name": "PHP: 7.4 WP: latest - 1",
```
Comment (adimoldovan) [inline plugins/woocommerce/package.json]: I wouldn't include the PHP version in the name if using the default, we'll probably forget to update it whenever the default version changes.

Comment (kalessil) [reply plugins/woocommerce/package.json]: Yep, reverted in ee2249f7e64d110a43afacca2be2bcca4454ffae

**Review by adimoldovan — APPROVED** [review]
👍🏼

**Pull request merged**